spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/service/impl/DefaultTaskService.java

```diff
@@ -171,75 +179,110 @@ public DefaultTaskService(DataSourceProperties dataSourceProperties,
+								taskNode.toExecutableDSL(), taskConfigurationProperties));
+				launchTaskDeploymentProperties = TaskServiceUtils.establishComposedTaskProperties(launchTaskDeploymentProperties,
+						taskNode);
+			}
+
+			AppRegistration appRegistration = this.registry.find(taskDefinition.getRegisteredAppName(),
+					ApplicationType.task);
+			Assert.notNull(appRegistration, "Unknown task app: " + taskDefinition.getRegisteredAppName());
+			Resource appResource = this.registry.getAppResource(appRegistration);
+			Resource metadataResource = this.registry.getAppMetadataResource(appRegistration);
+
+			TaskExecution taskExecution = taskExecutionRepository.createTaskExecution(taskName);
+			taskDefinition = TaskServiceUtils.updateTaskProperties(taskDefinition,
+					dataSourceProperties);
+
+			Map<String, String> appDeploymentProperties = new HashMap<>(commonApplicationProperties.getTask());
+			appDeploymentProperties.putAll(
+					TaskServiceUtils.extractAppProperties(taskDefinition.getRegisteredAppName(), launchTaskDeploymentProperties));
+
+			Map<String, String> deployerDeploymentProperties = DeploymentPropertiesUtils
+					.extractAndQualifyDeployerProperties(launchTaskDeploymentProperties, taskDefinition.getRegisteredAppName());
+			if (StringUtils.hasText(this.dataflowServerUri) && taskNode.isComposed()) {
+				TaskServiceUtils.updateDataFlowUriIfNeeded(this.dataflowServerUri, appDeploymentProperties,
+						commandLineArgs);
+			}
+			AppDefinition revisedDefinition = TaskServiceUtils.mergeAndExpandAppProperties(taskDefinition, metadataResource,
+					appDeploymentProperties, this.whitelistProperties);
+			List<String> updatedCmdLineArgs = this.updateCommandLineArgs(commandLineArgs, taskExecution);
+			AppDeploymentRequest request = new AppDeploymentRequest(revisedDefinition, appResource,
+					deployerDeploymentProperties, updatedCmdLineArgs);
+
+			RequestCriteria requestCriteria = new RequestCriteria(request, taskDefinition, taskExecution, launchTaskDeploymentProperties);
+			return requestCriteria;
+		});
+	}
+
+	private void postLaunchProcessing(RequestCriteria requestCriteria,
+			String externalExecutionId) {
+		TransactionTemplate template = new TransactionTemplate(transactionManager);
+
+		template.execute(new TransactionCallbackWithoutResult() {
+
+			@Override
+			protected void doInTransactionWithoutResult(TransactionStatus transactionStatus) {
+				TaskDefinition taskDefinition = requestCriteria.getTaskDefinition();
 
-		this.auditRecordService.populateAndSaveAuditRecordUsingMapData(
-				AuditOperationType.TASK, AuditActionType.DEPLOY,
-				taskDefinition.getName(), getAuditata(taskDefinition, taskDeploymentProperties, updatedCmdLineArgs));
+				taskExecutionRepository.updateExternalExecutionId(
+						requestCriteria.getTaskExecution().getExecutionId(),
+						externalExecutionId);
 
-		return taskExecution.getExecutionId();
+				auditRecordService.populateAndSaveAuditRecordUsingMapData(
```

Comment:
was it intentional that `AuditRecordService` implementation is not annotated with `@Transactional`?  @ghillert